merlin-server/src/test/java/gov/nasa/jpl/aerie/merlin/server/models/MissionModelTest.java

```diff
@@ -24,150 +24,153 @@ public final class MissionModelTest {
+  }
+
+  @Test
+  public void shouldInstantiateActivityInstance()
+  throws MissionModelService.NoSuchActivityTypeException, InvalidArgumentsException
+  {
+    // GIVEN
+    final var typeName = "foo";
+    final var parameters = new HashMap<>(Map.of("x", SerializedValue.of(0),
+                                                "y", SerializedValue.of("test"),
+                                                "z", SerializedValue.of(1)));
+
+    // WHEN
+    final var activity = new SerializedActivity(typeName, parameters);
+    final var specType = Optional
+        .ofNullable(registry.taskSpecTypes().get(activity.getTypeName()))
+        .orElseThrow(() -> new MissionModelService.NoSuchActivityTypeException(activity.getTypeName()));
+    final var failures = specType.validateArguments(activity.getArguments());
+
+    // THEN
+    assertThat(failures).isEmpty();
+  }
+
+  @Test
+  public void shouldNotInstantiateActivityInstanceWithIncorrectParameterType() {
+    // GIVEN
+    final var typeName = "foo";
+    final var parameters = new HashMap<>(Map.of("x", SerializedValue.of(0),
+                                                "y", SerializedValue.of(1.0)));
+
+    // WHEN
+    final var thrown = catchThrowable(() -> {
+        final var activity = new SerializedActivity(typeName, parameters);
+        final var specType = Optional
+            .ofNullable(registry.taskSpecTypes().get(activity.getTypeName()))
+            .orElseThrow(() -> new MissionModelService.NoSuchActivityTypeException(activity.getTypeName()));
+        specType.validateArguments(activity.getArguments());
+    });
 
-        // THEN
-        assertThat(type).isEqualTo(expectedType);
-    }
-
-    @Test
-    public void shouldNotGetActivityTypeForNonexistentActivityType() {
-        // GIVEN
-        final String activityId = "nonexistent activity type";
-
-        // WHEN
-        final var thrown = catchThrowable(() -> {
-            final var specType = Optional
-                .ofNullable(registry.taskSpecTypes().get(activityId))
-                .orElseThrow(() -> new MissionModelService.NoSuchActivityTypeException(activityId));
-
-            new ActivityType(
-                activityId,
-                specType.getParameters(),
-                specType.getRequiredParameters(),
-                specType.getReturnValueSchema());
-        });
-
-        // THEN
-        assertThat(thrown).isInstanceOf(MissionModelService.NoSuchActivityTypeException.class);
-    }
-
-    @Test
-    public void shouldInstantiateActivityInstance()
-    throws MissionModelService.NoSuchActivityTypeException, InvalidArgumentsException
-    {
-        // GIVEN
-        final var typeName = "foo";
-        final var parameters = new HashMap<>(Map.of("x", SerializedValue.of(0),
-                                                    "y", SerializedValue.of("test")));
-
-        // WHEN
+    // THEN
+    assertThat(thrown).isInstanceOf(InvalidArgumentsException.class);
```

Comment:
🔶 I think it would communicate the intent of this test better if we add a couple more assert statements here, showing that `thrown.missingArguments` contains `z`, and that `unconstructableArguments` contains `y`.

Reply:
Good idea! Will do

Reply:
Just pushed this change :)